torchtune/_cli/download.py

```diff
@@ -4,70 +4,101 @@
+            help="Download a model from the HuggingFace Hub.",
+            description="Download a model from the HuggingFace Hub.",
+            epilog=textwrap.dedent(
+                """\
+            examples:
+                # Download a model from the HuggingFace Hub with a Hugging Face API token
+                $ tune download meta-llama/Llama-2-7b-hf --hf-token <TOKEN> --output-dir /tmp/model
+                Successfully downloaded model repo and wrote to the following locations:
+                ./model/config.json
+                ./model/README.md
+                ./model/consolidated.00.pth
+                ...
 
-    Raises:
-        ValueError: If the model is not supported.
-    """
-    if "meta-llama" in repo_id and hf_token is None:
-        raise ValueError(
-            "You need to provide a Hugging Face API token to download gated models."
-            "You can find your token by visiting https://huggingface.co/settings/tokens"
-        )
+                # Download an ungated model from the HuggingFace Hub
+                $ tune download mistralai/Mistral-7B-Instruct-v0.2
+                Successfully downloaded model repo and wrote to the following locations:
+                ./model/config.json
+                ./model/README.md
+                ./model/model-00001-of-00003.safetensors
+                ...
 
-    # Download the tokenizer and PyTorch model files
-    snapshot_download(
-        repo_id,
-        local_dir=output_dir,
-        resume_download=True,
-        token=hf_token,
-    )
+            For a list of all models, visit the HuggingFace Hub https://huggingface.co/models.
+            """
+            ),
+            formatter_class=argparse.RawTextHelpFormatter,
+        )
+        self._add_arguments()
+        self._parser.set_defaults(func=self._download_cmd)
 
-    print(
-        "Succesfully downloaded model repo and wrote to the following locations:",
-        *list(output_dir.iterdir()),
-        sep="\n",
-    )
+    def _add_arguments(self) -> None:
+        """Add arguments to the parser."""
+        self._parser.add_argument(
+            "repo_id",
+            type=str,
+            help="Name of the repository on HuggingFace Hub.",
+        )
+        self._parser.add_argument(
+            "--output-dir",
+            type=Path,
+            required=False,
+            default="./model",
+            help="Directory in which to save the model.",
+        )
+        self._parser.add_argument(
+            "--hf-token",
+            type=str,
+            required=False,
+            default=os.getenv("HF_TOKEN", None),
+            help="HuggingFace API token. Needed for gated models like Llama2.",
+        )
 
+    def _download_cmd(self, args: argparse.Namespace) -> None:
+        """Downloads a model from the HuggingFace Hub."""
+        # Download the tokenizer and PyTorch model files
+        try:
+            true_output_dir = snapshot_download(
```

Comment:
Can we add "ignore_patterns" and set it to by default ignore "*.safetensors"? I need to make this change everytime I run on runpod and it's really annoying. If a model ONLY has safetensors, the user can manually set it to an empty string. Just print out that you're ignoring this for space. It's also really annoying because the saved files are cached by default and I need to dig into how to get rid of something I don't want (unless I'm just being stupid).

Reply:
Here's the reference: https://huggingface.co/docs/huggingface_hub/en/guides/download

Reply:
Sure I was gunna do this in a follow-up, but can do it here. 

In the future, I actually think we might want to by default support safetensors, but that's a discussion for another time.